ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Protocol.hs

```diff
@@ -112,6 +115,13 @@ instance CanHardFork xs => ConsensusProtocol (HardForkProtocol xs) where
   -- Security parameter must be equal across /all/ eras
   protocolSecurityParam = hardForkConsensusConfigK
 
+  protocolSecurityParamConsistencyCheck HardForkConsensusConfig {..} = do
+    let allSecurityParams = hardForkConsensusConfigK :|
+          perEraConsensusConfigSecurityParams hardForkConsensusConfigPerEra
+    case nub allSecurityParams of
```

Comment:
`nub` is quadratic; use eg `Data.Set` instead

Reply:
removed, don't need a `nub` effect at all